stackrox/stackrox
# ROX-14603: Do not touch databases on legacy PVC in 74 to 4.0 upgrade

## Description

After RocksDB to Postgres migration, we require customer to upgrade from 3.74. The database on stackrox-db PVC will not be used any further after 4.0 but we want to keep it there for further recovery.
1. Rollback to previous version if there is a previous version on stackrox-db PVC
2. Switch back to its version before upgrade.

This PR makes the change to keep migrated from 3.74 legacy DB intacted on legacy PVC.

Note: This diff is based on cong/fix which contains a few unit test fixes.

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

Manual test is required:
1. Upgrade from 3.74 to current check databases
2. Rollback to 3.74 by switching the version
3. Remove central-db
4. Upgrade to 3.74 again.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.
